File: Java/src/_0401_Binary_Watch/Solution.java
package _0401_Binary_Watch;

import java.util.ArrayList;
import java.util.List;

/**
 * Date: 2021/5/31 15:13
 * 执行用时：13 ms, 在所有 Java 提交中击败了28.21%的用户
 * 内存消耗：38.4 MB, 在所有 Java 提交中击败了35.46%的用户
 */
public class Solution {

    public List<String> readBinaryWatch(int n) {
        List<String> res = new ArrayList<>();
        for (int i = 0; i < 1 << 10; i++) {
            int s = 0;
            for (int j = 0; j < 10; j++)
                if ((i >> j & 1) == 1)
                    s++;
            if (s == n) {
                int a = i >> 6, b = i & 63;
                if (a < 12 && b < 60) res.add(a + ":" + (b < 10 ? "0" + b : "" + b));
            }
        }
        return res;
    }

    public static void main(String[] args) {

        System.out.println((new Solution()).readBinaryWatch(1));
    }
}

File: Python/_0401_Binary_Watch/Solution.py
# coding=utf-8
# Date: 2021/6/21 11:01
from typing import List


# 执行用时：60 ms, 在所有 Python3 提交中击败了11.52%的用户
# 内存消耗：14.9 MB, 在所有 Python3 提交中击败了72.35%的用户
class Solution:
    def readBinaryWatch(self, turnedOn: int) -> List[str]:
        res = []
        for i in range(1 << 10):
            s = 0
            for j in range(10):
                if i >> j & 1:
                    s += 1
            if s == turnedOn:
                a = i >> 6
                b = i & 63
                if a < 12 and b < 60:
                    res.append(str(a) + ":" + ("0" if b < 10 else "") + str(b))
        return res


if __name__ == "__main__":
    print(Solution().readBinaryWatch(1))
